console/console-init/ui/src/modules/iot-device/components/CreateMetadata/MetaDataRow.tsx

```diff
@@ -12,28 +12,42 @@ import { MetaDataType } from "./MetaDataType";
 import { MetaDataValue } from "./MetaDataValue";
 
 export interface IMetaDataRowProps {
+  metadataRow: any;
   metadataList: any;
-  setMetadataList: (metadataList: any) => void;
-  rowIndex: number;
+  setMetadataList: (metadataRow: any) => void;
+  rowId: string;
 }
 
 export const MetaDataRow: React.FC<IMetaDataRowProps> = ({
+  metadataRow,
   metadataList,
   setMetadataList,
-  rowIndex
+  rowId
 }) => {
   const [validationStatus, setValidationStatus] = useState<
     ValidationStatusType.DEFAULT | ValidationStatusType.ERROR
   >(ValidationStatusType.DEFAULT);
 
   const updateMetadataList = (property: string, value: string) => {
     let updatedMetadataList = [...metadataList];
-    updatedMetadataList[rowIndex][property] = value;
+    const index = searchMetadataById(rowId);
+    updatedMetadataList[index][property] = value;
     if (property === "type" && isObjectOrArray(value as any))
-      updatedMetadataList[rowIndex].value = "";
+      updatedMetadataList[0].value = [];
```

Comment:
is there any reason for having hard code index `0` , updatedMetadataList[0].value